src/bench/verification.cpp

```diff
@@ -49,7 +49,9 @@ static void ECDSA(benchmark::State& state)
         mtx.vout[i].scriptPubKey = CScript() << OP_1;
     }
 
-    const PrecomputedTransactionData txdata(mtx);
+    std::vector<CTxOut> allPrevOutputs;
+    allPrevOutputs.resize(mtx.vin.size());
+    const PrecomputedTransactionData txdata(mtx, allPrevOutputs);
```

Comment:
I don't understand this, you're passing in a default-initialized vector of the prevouts? It would be helpful to have a comment here describing why this is okay (presumably because it's benchmarking code) but more broadly I find the fact that it's possible to pass such a "faked" value a little worrying. It means that every call site has to be inspected with extreme care to ensure that it's not possible to bypass sending in invalid data.

Reply:
"Faked data" just means we're committing to the wrong inputs (in this case, zero-valued `CTxOut`s with empty `scriptPubKey`s). The only place we need to ensure we commit to the correct inputs is the consensus validation logic. Everywhere else, if we commit to the wrong inputs then the transaction just won't get mined.